src/Catalog/Registration/FlatContainerPackagePathProvider.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using NuGet.Services.Metadata.Catalog.Helpers;
+
+namespace NuGet.Services.Metadata.Catalog.Registration
+{
+    public class FlatContainerPackagePathProvider : IPackagePathProvider
+    {
+        public string GetPackagePath(string id, string version)
+        {
+            version = NuGetVersionUtility.NormalizeVersion(version);
+
+            return $"v3-flatcontainer/{id.ToLowerInvariant()}/{version.ToLowerInvariant()}/{id.ToLowerInvariant()}.{version.ToLowerInvariant()}.nupkg";
```

Comment:
This should use the V3 index.json to determine the base address of the flat-container. There's not guarantee that we will always use `v3-flatcontainer` or that we will only have a single flat-container.